### ISSUE TITLE
00-eda-airbnb-ny.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/alemantellini-machine-learning#1]
Hola @alemantellini  felicidades por tu trabajo. Has hecho un **excelente análisis exploratorio** con buenas visualizaciones y conclusiones. Sin embargo, hay algún error a tener en cuenta:

---

### **Error 1: Data Leakage en Scaling**

En tu código actual, haces scaling **antes** de hacer el split train/test:

**ACTUAL:**
```python
scaler = MinMaxScaler()
scaled = scaler.fit_transform(total_data[variables])  # ← Fit sobre TODOS los datos
df_scaler = pd.DataFrame(scaled, columns=variables, index=total_data.index)
df_scaler["price"] = total_data["price"]
```

**PROBLEMA:**
El scaler aprende los valores mín/máx de **TODO el dataset**, incluyendo datos que deberían ser desconocidos (test set). Esto causa **data leakage**: el modelo aprenderá información del test set antes incluso de entrenar.

---

### **CORRECCIÓN:**

El orden correcto es:
1. **Split primero** (train/test)
2. **Fit el scaler SOLO en train**
3. **Aplicar (transform) a train y test por separado**

```python
# 1️⃣ PRIMERO: Preparar features y target
mi_x = total_data[variables]
mi_y = total_data["price"]

# 2️⃣ SPLIT (ANTES de scaling)
X_train, X_test, y_train, y_test = train_test_split(
    mi_x, mi_y, test_size=0.2, random_state=42
)

# 3️⃣ SCALING - Fit SOLO en train
scaler = MinMaxScaler()
scaler.fit(X_train)  # ← Fit SOLO en train

# 4️⃣ Transform en train y test por separado
X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

# Convertir a DataFrame si lo necesitas
X_train_scaled = pd.DataFrame(X_train_scaled, columns=variables)
X_test_scaled = pd.DataFrame(X_test_scaled, columns=variables)
```

---

### **Nota: No eliminas outliers detectados**

No hace falta si lo haces ante decisión, decirte que identificaste bien los outliers con IQR, pero **no los eliminas antes del scaling**. Los boxplots muestran claramente outliers extremos, especialmente en `price` y `minimum_nights` si es parte de tu decisión no quitarlos está bien, si no lo es se te ha pasado, no es un error en si, simplemente di que decides hacer:

**ACTUAL (INCOMPLETO):**
```python
# Identificas outliers pero no haces nada con ellos...
price_iqr = price_stats["75%"] - price_stats["25%"]
upper_limit = price_stats["75%"] + 1.5 * price_iqr
lower_limit = price_stats["25%"] - 1.5 * price_iqr
# Fin del análisis, pero no eliminas
```

**CORRECCIÓN:**

Después de identificar los límites, **elimina los outliers** o di **que has decidido no eliminarlos**:

```python
# Para price
price_iqr = price_stats["75%"] - price_stats["25%"]
upper_limit = price_stats["75%"] + 1.5 * price_iqr
lower_limit = price_stats["25%"] - 1.5 * price_iqr

# Eliminar outliers de price
total_data = total_data[(total_data["price"] >= lower_limit) & 
                         (total_data["price"] <= upper_limit)]

# Para minimum_nights (si tiene muchos outliers)
nights_iqr = nights_stats["75%"] - nights_stats["25%"]
upper_limit = nights_stats["75%"] + 1.5 * nights_iqr
total_data = total_data[total_data["minimum_nights"] <= upper_limit]

print(f"Dataset después de eliminar outliers: {total_data.shape}")
```

---

### **Resumen: Orden correcto del flujo**

```
1. Carga de datos
   ↓
2. Análisis exploratorio (✅ Hecho perfecto)
   ↓
3. Limpieza: eliminar nulos, columnas irrelevantes
   ↓
4. ⚠️ ELIMINAR OUTLIERS (Aquí falta)
   ↓
5. ⚠️ SPLIT TRAIN/TEST (ANTES de scaling - Aquí estaba en el orden incorrecto)
   ↓
6. ⚠️ SCALING SOLO en TRAIN, luego aplicar a TEST
   ↓
7. Feature Selection (✅ Lo hiciste correctamente después del split)
   ↓
8. Guardar datasets (✅ Bien hecho)
```

---

### **¿Por qué es importante esto?**

Si mantienes el scaling sobre todos los datos, un modelo entrenado con tus datos **parecerá tener mejor rendimiento de lo que realmente tiene**. En competiciones o en producción, fall fallará porque el scaler nunca "vio" los datos reales del test.

**Tu objetivo:** Que el modelo generalice bien. El test set debe simular datos "nuevos" que el modelo nunca vio.
